viadot/sources/databricks.py

```diff
@@ -0,0 +1,396 @@
+    ):
+        """
+        Insert data into a table.
+
+        Args:
+            table_name (str): Name of the table to which data will be inserted.
+            df (pd.DataFrame): DataFrame with the data to be inserted into the table.
+            primary_key (str, Optional): Needed only when updating the data, the primary key on which the data will be joined.
+            if_exists (str, Optional): Which operation to run with the data. Allowed operations are: replace, append and update.
+
+        Example:
+        ```python
+        from viadot.sources import Databricks
+        databricks = Databricks()
+        list = [{"id":"1", "name":"Joe"}]
+        df = pd.DataFrame(list)
+        databricks.insert_into(table_name="raw.c4c_test4", df=df, primary_key="pk", if_exists="update")
+        ```
+        """
+        IF_EXISTS_ACCEPTED_VALUES = ["replace", "append", "update", "fail"]
+        if if_exists not in IF_EXISTS_ACCEPTED_VALUES:
+            raise ValueError()
+        if self._check_if_table_exists(table_name) and if_exists == "fail":
+            raise ValueError()
+        if if_exists == "replace":
+            self._full_refresh(table_name, df)
+        elif if_exists == "append":
+            self._append(table_name, df)
+        else:
+            self._upsert(table_name, primary_key, df)
+
+    def create_schema(self, schema_name: str):
+        """
+        Create a schema for storing tables.
+
+        Args:
+            schema_name (str): Name of the new schema to be created.
+
+        Example:
+        ```python
+        from viadot.sources import Databricks
+        databricks = Databricks()
+        databricks.create_schema("schema_1")
+        ```
+        """
+        self.session.sql(f"CREATE SCHEMA IF NOT EXISTS {schema_name}")
+        logger.info(f"Schema {schema_name} created.")
+
+    def delete_schema(self, schema_name: str):
+        """
+        Delete a schema.
+
+        Args:
+            schema_name (str): Name of the schema to be deleted.
+
+        Example:
+        ```python
+        from viadot.sources import Databricks
+        databricks = Databricks()
+        databricks.delete_schema("schema_1")
+        ```
+        """
+        self.session.sql(f"DROP SCHEMA {schema_name}")
+        logger.info(f"Schema {schema_name} deleted.")
+
+    # TODO: Change to return dict of col_name:dtype
+    def discover_schema(self, table_name: str) -> dict:
+        """
+        Return a table's schema.
+
+        Args:
+            table_name (str): Name of the table whose schema will be printed.
+
+        Example:
+        ```python
+        from viadot.sources import Databricks
+        databricks = Databricks()
+        databricks.discover_schema("table_1")
+        ```
+        Returns:
+            schema: A dictionary containing the schema details of the table.
+        """
+        result = self.run(f"DESCRIBE {table_name}", "dataframe")
```

Comment:
It seems like you make your life harder by fetching this is a DataFrame, can't we fetch this as a list or dict?

Reply:
Unfortunately no, since it's a Spark sql statement running, it's not possible to directly fetch it as a list or dict.

Reply:
Then why do we have the `fetch_type` parameter at all? 😅 If you don't use it then you can remove it. Then this stays the same just without having to pass "dataframe" as the argument to `fetch_type`.

Reply:
I'm not sure what you mean. By default, it fetches it as a Spark DataFrame (the default record type), but I want it to be returned as a Pandas DataFrame for the ease of manipulating it.